zooniverse/front-end-monorepo
# Add reset to TextFromSubjectTask

## Package
- lib-classifier

## Linked Issue and/or Talk Post
- towards InVision - [TextFromSubject Task with Reset](https://projects.invisionapp.com/d/main/default/#/console/21704442/465346191/preview)

## Describe your changes
- add `resetToSubject` action to TextFromSubject task annotation
- add Reset button to TextFromSubject task component
- add `isChanged` view to TextFromSubject task annotation, to disable Reset button if annotation value is unchanged from subject text location content
- update Reset button for `isChanged` annotation view

Note - I'll be converting all of the TextFromSubject task tests that use enzyme to use RTL in a separate PR

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - https://local.zooniverse.org:8080/?project=1952&workflow=3602&demo=true&env=staging
- What user actions should my reviewer step through to review this PR?
  1. note the initial text input value
  2. edit the text input value
  3. click reset
  4. text input value should be back to initial value (step 1)
- Which storybook stories should be reviewed?
  - http://localhost:6006/?path=/story/tasks-textfromsubject--default&globals=locale:en;locales.en:English;locales.test:Test+Language
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes, a confirmation modal component, see #3044 

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)
